contracts/src/v0.8/VRFConsumerBaseV2Upgradable.sol

```diff
@@ -0,0 +1,143 @@
+ * @dev if your contract could have multiple requests in flight simultaneously.
+ *
+ * @dev Colliding `requestId`s are cryptographically impossible as long as seeds
+ * @dev differ.
+ *
+ * *****************************************************************************
+ * @dev SECURITY CONSIDERATIONS
+ *
+ * @dev A method with the ability to call your fulfillRandomness method directly
+ * @dev could spoof a VRF response with any random value, so it's critical that
+ * @dev it cannot be directly called by anything other than this base contract
+ * @dev (specifically, by the VRFConsumerBase.rawFulfillRandomness method).
+ *
+ * @dev For your users to trust that your contract's random behavior is free
+ * @dev from malicious interference, it's best if you can write it so that all
+ * @dev behaviors implied by a VRF response are executed *during* your
+ * @dev fulfillRandomness method. If your contract must store the response (or
+ * @dev anything derived from it) and use it later, you must ensure that any
+ * @dev user-significant behavior which depends on that stored value cannot be
+ * @dev manipulated by a subsequent VRF request.
+ *
+ * @dev Similarly, both miners and the VRF oracle itself have some influence
+ * @dev over the order in which VRF responses appear on the blockchain, so if
+ * @dev your contract could have multiple VRF requests in flight simultaneously,
+ * @dev you must ensure that the order in which the VRF responses arrive cannot
+ * @dev be used to manipulate your contract's user-significant behavior.
+ *
+ * @dev Since the block hash of the block which contains the requestRandomness
+ * @dev call is mixed into the input to the VRF *last*, a sufficiently powerful
+ * @dev miner could, in principle, fork the blockchain to evict the block
+ * @dev containing the request, forcing the request to be included in a
+ * @dev different block with a different hash, and therefore a different input
+ * @dev to the VRF. However, such an attack would incur a substantial economic
+ * @dev cost. This cost scales with the number of blocks the VRF oracle waits
+ * @dev until it calls responds to a request. It is for this reason that
+ * @dev that you can signal to an oracle you'd like them to wait longer before
+ * @dev responding to the request (however this is not enforced in the contract
+ * @dev and so remains effective only in the case of unmodified oracle software).
+
+ * @dev The VRFConsumerBaseV2Upgradable is an upgradable variant of VRFConsumerBaseV2
+ * @dev (see https://docs.openzeppelin.com/upgrades-plugins/1.x/writing-upgradeable).
+ * @dev It's semantics are identical to VRFConsumerBaseV2 and can be inherited from
+ * @dev to create an upgradeable VRF consumer contract.
+ */
+
+import "@openzeppelin/contracts-upgradeable/proxy/utils/Initializable.sol";
+
+abstract contract VRFConsumerBaseV2Upgradeable is Initializable {
+  error OnlyCoordinatorCanFulfill(address have, address want);
+  address private vrfCoordinator;
+
+  /**
```

Comment:
We should add a gap for upgrades `uint256[50] private ______gap;`

Reply:
Each uint256 will cover a single storage slot, so for example when adding stuff in the future two uint128s added would result in one slot fewer. https://docs.soliditylang.org/en/latest/internals/layout_in_storage.html
```
uint128 a
uint128 b
uint256[49] private ______gap
```

Reply:
Sounds good, will look into this.